pkg/scheduler/objects/queue.go

```diff
@@ -1067,6 +1073,11 @@ func (sq *Queue) TryAllocate(iterator func() NodeIterator) *Allocation {
 		headRoom := sq.getHeadRoom()
 		// process the apps (filters out app without pending requests)
 		for _, app := range sq.sortApplications(true) {
+			if !app.IsRunning() && !app.queue.canRun() {
```

Comment:
You are in the queue, should not access the app for this info. It also does not change while walking over the application list. You can pre-calculate the value and use it.